app/controllers/concerns/mission_control/jobs/job_filters.rb

```diff
@@ -9,10 +9,16 @@ module MissionControl::Jobs::JobFilters
 
   private
     def set_filters
-      @job_filters = { job_class_name: params.dig(:filter, :job_class_name).presence, queue_name: params.dig(:filter, :queue_name).presence }.compact
+      @job_filters = { job_class_name: params.dig(:filter, :job_class_name).presence, queue_name: params.dig(:filter, :queue_name).presence,
+                       finished_at: date_with_time_zone(params.dig(:filter, :finished_at_start).presence)..date_with_time_zone(params.dig(:filter, :finished_at_end).presence) }.compact
     end
 
     def active_filters?
       @job_filters.any?
     end
+
+    def date_with_time_zone(date)
+      return date if date.nil?
+      DateTime.parse(date).in_time_zone
```

Comment:
```suggestion
      if date.present?
        DateTime.parse(date).in_time_zone
      end
```

I'd rewrite this in this way, and you can remove the `presence` calls above.